website/addons/googledrive/views/config.py

```diff
@@ -34,41 +35,65 @@ def googledrive_config_get(node_addon, auth, **kwargs):
 def googledrive_config_put(node_addon, auth, **kwargs):
     """View for changing a node's linked Google Drive folder/file."""
     selected = request.get_json().get('selected')
-    node_addon.set_folder(selected, auth=auth)
+    node_addon.set_target_folder(selected, auth=auth)
     node_addon.save()
     return {
         'result': {
             'folder': {
                 'name': selected['path'],
             },
-            'urls': serialize_urls(node_addon),
+            'urls': GoogleDriveSerializer(
+                node_settings=node_addon,
+                user_settings=auth.user.get_addon('googledrive')
+            ).serialized_urls
         },
         'message': 'Successfully updated settings.',
     }
 
-
 @must_be_logged_in
 def googledrive_user_config_get(auth, **kwargs):
     """View for getting a JSON representation of the logged-in user's
     Google Drive user settings.
     """
-    urls = {
-        'create': api_url_for('googledrive_oauth_start_user'),
-        'delete': api_url_for('googledrive_oauth_delete_user'),
-    }
+    user_settings = auth.user.get_addon('googledrive')
+    return GoogleDriveSerializer(user_settings=user_settings).serialized_user_settings
 
-    user_addon = auth.user.get_addon('googledrive')
 
-    user_has_auth = False
-    username = ''
-    if user_addon:
-        user_has_auth = user_addon.has_auth
-        username = user_addon.username
+@must_have_permission(permissions.WRITE)
+@must_have_addon('googledrive', 'node')
+def googledrive_import_user_auth(auth, node_addon, **kwargs):
```

Comment:
Ok then, if there will never be a case where this endpoint is reached and the user does not have a connected account then it should be fine.